wicker/core/datasets.py

```diff
@@ -114,6 +126,57 @@ def iterate_bucket_key_chunk_for_size(bucket_key_locs: List[Tuple[str, str]]) ->
 class AbstractDataset(abc.ABC):
     """Interface for a Map-style (non-streaming) Dataset"""
 
+    def __init__(
+        self,
+        dataset_name: str,
+        dataset_partition_name: str,
+        dataset_version: str,
+        path_factory: WickerPathFactory,
+        pa_filesystem: pafs.FileSystem,
+        storage: AbstractDataStorage,
+        columns_to_load: Optional[List[str]] = None,
+        filelock_timeout_seconds: int = FILE_LOCK_TIMEOUT_SECONDS,
```

Comment:
This is not so abstract, kind of concrete cache implementation detail, could we make a CacheManger Interface and inject into the dataset just as the PathFactory etc?

Reply:
I agree in principle but it's unfortunately out of scope for this PR. It would be nice to abstract it away further and more easily unify the interface. At current though I don't want to take on the work although the cache improvements in the PR get you mostly there with it's abstraction of the reading. If it's okay with you, I'll make an issue and you can make a request to me or @convexquad in the future to do this?

Reply:
@zhenyu I totally agree with you! In the original code at https://github.com/woven-planet/wicker/blob/main/wicker/core/datasets.py#L141 the constructor of the original `S3Dataset` class exposes too many implementation details that should be wrapped up in another class.

However, in this PR we are not trying to fix the design. If it is ok with you, we will only try to enable the local storage correctly (without breaking S3 storage) and we will not try to refactor the design yet. I think if we can show that with these changes that the data loading from local disk and S3 are both working correctly, then it will be perfect time to refactor the class design here.